src/librustc_resolve/lib.rs

```diff
@@ -2988,6 +3007,7 @@ impl<'a> Resolver<'a> {
                                               ResolutionError::SelfNotAvailableInStaticMethod);
                             } else {
                                 let last_name = path.segments.last().unwrap().identifier.name;
+                                let mut is_field = false;
```

Comment:
I'd be better than using a mutable variable. This way you are sure that you cannot forget to specify it in any `match` case, even match cases we could add in the future.